platforms/jvm/scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompilerFactory.java

```diff
@@ -63,31 +62,26 @@ public class ZincScalaCompilerFactory {
     private static final int COMPILER_CLASSLOADER_CACHE_SIZE = 4;
     private static final String SCALA_3_COMPILER_ID = "scala3-compiler_3";
     private static final String SCALA_3_LIBRARY_ID = "scala3-library_3";
-    private static final GuavaBackedClassLoaderCache<HashCode> CLASSLOADER_CACHE = new GuavaBackedClassLoaderCache<HashCode>(CLASSLOADER_CACHE_SIZE);
+    private static final SimpleBackedClassLoaderCache<HashCode> CLASSLOADER_CACHE = new SimpleBackedClassLoaderCache<HashCode>(CLASSLOADER_CACHE_SIZE);
     private static final ClassLoaderCache COMPILER_CLASSLOADER_CACHE;
 
     static {
-        // Load TimeCheckingClassLoaderCache and use it to create cache via reflection
-        // If we detect that we are using zinc 1.2.x, we fallback to default cache
-        Class<?> abstractCacheClass;
-        Class<?> checkingClass;
+        ClassLoaderCache cache;
         try {
-            abstractCacheClass = ZincScalaCompilerFactory.class.getClassLoader().loadClass("sbt.internal.inc.classpath.AbstractClassLoaderCache");
-            checkingClass = ZincScalaCompilerFactory.class.getClassLoader().loadClass("org.gradle.api.internal.tasks.scala.TimeCheckingClassLoaderCache");
+            cache = createCompilerClassloaderCache();
         } catch (ClassNotFoundException ex) {
-            abstractCacheClass = null;
-            checkingClass = null;
+            cache = new ClassLoaderCache(new URLClassLoader(new URL[]{}));
         }
-        if (checkingClass != null) {
-            try {
-                Constructor<ClassLoaderCache> constructor = ClassLoaderCache.class.getConstructor(abstractCacheClass);
-                Object cache = checkingClass.getConstructors()[0].newInstance(COMPILER_CLASSLOADER_CACHE_SIZE);
-                COMPILER_CLASSLOADER_CACHE = constructor.newInstance(cache);
-            } catch (Exception e) {
-                throw new RuntimeException("Failed to instantiate ClassLoaderCache", e);
-            }
-        } else {
-            COMPILER_CLASSLOADER_CACHE = new ClassLoaderCache(new URLClassLoader(new URL[]{}));
+        COMPILER_CLASSLOADER_CACHE = cache;
+    }
+
+    private static ClassLoaderCache createCompilerClassloaderCache() throws ClassNotFoundException {
+        // Load TimeCheckingClassLoaderCache and use it to create cache
+        // If we detect that we are using zinc 1.2.x, we fallback to default cache
+        try {
+            return new ClassLoaderCache(new TimeCheckingClassLoaderCache(COMPILER_CLASSLOADER_CACHE_SIZE));
+        } catch (Exception e) {
```

Comment:
I think this isn't correct because this turns even ClassNotFoundExceptions into runtime exceptions, but I'm suspicious if we have any test coverage for this